### PR TITLE
feat: betterperformancelogic

### PR DESCRIPTION
**Pull request by Decodetalkers**
Issue: https://github.com/linuxdeepin/developer-center/issues/6514 Log:

### CODE REVIEW

**Comment by Decodetalkers** [comment]
@xzl

File: system/power1/cpu_handler.go [hunk 1]
```diff
@@ -340,6 +339,22 @@ func (cpus *CpuHandlers) getCpuGovernorPath(isPstate bool) string {
 	return path
 }
 
+func cpuHasPstate() bool {
+	if !dutils.IsFileExist(globalDefaultDriverPath) {
+		return false
+	}
+	driverdata, err := os.ReadFile(globalDefaultDriverPath)
```
Comment (dengbo11) [inline system/power1/cpu_handler.go:346]: driverData

File: system/power1/cpu_handler_test.go [hunk 2]
```diff
@@ -5,11 +5,25 @@
...
+	"os"
+	"testing"
 )
 
+func Test_IsPstate(t *testing.T) {
+	driverdata, _ := os.ReadFile("./testdata/scaling_driver_intel_pstate")
+	driver := string(driverdata)
+	assert.Equal(t, driverHasPstate(driver), true)
+
+	driverdata, _ = os.ReadFile("./testdata/scaling_driver_amd_pstate")
```
Comment (dengbo11) [inline system/power1/cpu_handler_test.go:18]: 同上

**Review by dengbo11 — APPROVED** [review]

**Review by dengbo11 — APPROVED** [review]

**Pull request merged**